pkg/querybuilder/where_clause_visitor.go

```diff
@@ -638,28 +640,38 @@ func (v *filterExpressionVisitor) VisitFunctionCall(ctx *grammar.FunctionCallCon
 	for _, key := range keys {
 		var fieldName string
 
-		if strings.HasPrefix(key.Name, v.jsonBodyPrefix) {
-			fieldName, _ = v.jsonKeyToKey(context.Background(), key, qbtypes.FilterOperatorUnknown, value)
+		if functionName == "hasToken" {
+			// this will only work with string.
+			if _, ok := value[0].(string); !ok {
+				v.errors = append(v.errors, fmt.Sprintf("function `%s` expects value parameter to be a string", functionName))
+				return ""
+			}
+			conds = append(conds, fmt.Sprintf("hasToken(LOWER(%s), LOWER(%s))", key.Name, v.builder.Var(value[0])))
 		} else {
-			// TODO(add docs for json body search)
-			if v.mainErrorURL == "" {
-				v.mainErrorURL = "https://signoz.io/docs/userguide/search-troubleshooting/#function-supports-only-body-json-search"
+			// this is that all other functions only support array fields
+			if strings.HasPrefix(key.Name, v.jsonBodyPrefix) {
+				fieldName, _ = v.jsonKeyToKey(context.Background(), key, qbtypes.FilterOperatorUnknown, value)
+			} else {
+				// TODO(add docs for json body search)
+				if v.mainErrorURL == "" {
+					v.mainErrorURL = "https://signoz.io/docs/userguide/search-troubleshooting/#function-supports-only-body-json-search"
```

Comment:
will have to update this, once the implementation is fine I will update the docs